pkg/services/publicdashboards/api/middleware.go

```diff
@@ -28,14 +28,14 @@ func SetPublicDashboardOrgIdOnContext(publicDashboardService publicdashboards.Se
 	}
 }
 
-// Adds public dashboard flag on context
+// SetPublicDashboardFlag Adds public dashboard flag on context
 func SetPublicDashboardFlag(c *models.ReqContext) {
 	c.IsPublicDashboardView = true
 }
 
-// Middleware to enforce that a public dashboards exists before continuing to
-// handler
-func RequiresValidAccessToken(publicDashboardService publicdashboards.Service) func(c *models.ReqContext) {
+// RequiresExistingAccessToken Middleware to enforce that a public dashboards exists before continuing to handler. This
+// method will query the database to ensure that
```

Comment:
Let's add a comment on when to use this.

"Use when we want to enforce a public dashboard is valid on an endpoint we do not maintain" 

or similar

Reply:
Fixed in f03e6246294081841db05abdd9d749c58a480306